docs/FATE_On_Spark_With_Pulsar.md

```diff
@@ -0,0 +1,338 @@
+
+The following operation will assume that the certificate of "proxy.fate.org" has been generated.
+
+#### Deploy Apache Traffic Server
+
+##### Install Apache Traffic Server
+
+1. Log in to the "proxy.fate.org" host and prepare the dependencies according to this [document](https://github.com/apache/trafficserver/tree/9.0.0).
+
+2. Download
+
+``` bash
+$ wget https://apache.claz.org/trafficserver/trafficserver-9.0.0.tar.bz2
+```
+
+3. Install
+
+``` bash
+$ mkdir /opt/ts
+$ tar xf trafficserver-9.0.0.tar.bz2
+$ cd trafficserver-9.0.0
+$ ./configure --prefix /opt/ts
+$ make                   
+$ make install
+$ echo 'export LD_LIBRARY_PATH=$LD_LIBRARY_PATH:/opt/ts/lib' >> ~/.profile
+$ source ~/.profile
+```
+
+Once the process was finished, the traffic server will be installed in the "/opt/ts" directory and the configuration file path is "/opt/ts/etc/trafficserver/".
+
+##### Start Apache Traffic Server
+
+1. Setting Configuration
+ - /opt/ts/etc/trafficserver/records.config
+ ```
+ CONFIG proxy.config.http.cache.http INT 0
+ CONFIG proxy.config.reverse_proxy.enabled INT 0
+ CONFIG proxy.config.url_remap.remap_required INT 0
+ CONFIG proxy.config.url_remap.pristine_host_hdr INT 0
+ CONFIG proxy.config.http.response_server_enabled INT 0
+
+ CONFIG proxy.config.http.server_ports STRING 8080 8080:ipv6 443:ssl
+
+ CONFIG proxy.config.http.connect_ports STRING 443 6650-6660
+
+ CONFIG proxy.config.ssl.CA.cert.filename STRING ca.cert.pem
+ CONFIG proxy.config.ssl.CA.cert.path STRING /opt/proxy
+
+ CONFIG proxy.config.ssl.server.cert.path STRING /opt/proxy
+ ```
+ - /opt/ts/etc/trafficserver/ssl_multicert.config
+ ```
+ dest_ip=* ssl_cert_name=proxy.cert.pem ssl_key_name=proxy.key.pem
+ ```
+ - /opt/ts/etc/trafficserver/sni.config This is a routing table, and the Proxy will forward the client's request to the address specified by "tunnel_route".
+
+ ``` bash
+ sni:
+  - fqdn: 10000.fate.org
+    tunnel_route: 192.168.0.2:6651
+  - fqdn: 9999.fate.org
+    tunnel_route: 192.168.0.3:6651
+ ```
+ 
+ For a more detailed description of the configuration file, please refer to the official [documents](https://docs.trafficserver.apache.org/en/9.0.x/admin-guide/configuring-traffic-server.en.html).
+ 
+ 2. Start the service
+ 
+Copy the certificate, private key, and CA certificate generated for ATS in the previous step (under the "proxy.fate.org" directory) to the host's "/opt/proxy" directory, and use the following command to start ATS:
+
+``` bash
+/opt/ts/bin/trafficserver start
+```
+
+#### Deploy Pulsar
+
+The deployment of Pulsar is described in detail in [pulsar_deployment_guide.md](https://github.com/FederatedAI/FATE/blob/develop-1.6-pulsar/cluster-deploy/doc/fate_on_spark/pulsar_deployment_guide.md). A user only need to add a certificate for the broker and open the security service port. The specific operations are as follows:
```

Comment:
The deployment of Pulsar is described in detail in -> The details of how Pulsar deploy can be find in